yt_dlp/extractor/floatplane.py

```diff
@@ -0,0 +1,137 @@
+from .common import InfoExtractor
+
+from ..utils import (
+    UnsupportedError,
+    parse_iso8601,
+    unified_strdate,
+    traverse_obj
+)
+
+
+class FloatplaneIE(InfoExtractor):
+    _VALID_URL = r'(?:https?://(?:www\.)?floatplane\.com/post/(?P<id>[a-zA-Z0-9-]+))'
+    _TESTS = [{
+        'url': 'https://www.floatplane.com/post/DdShecU983',
+        'info_dict': {
+            'id': 'DdShecU983',
+            'ext': 'mp4',
+            'title': 'TQ: Are Intel Arc Graphics A Bad Idea?',
+            'thumbnail': r're:^https?://.*\.jpg$',
+        },
+        'skip': 'Requires premium Floatplane account',
+    }]
+
+    def _real_initialize(self):
+        self._download_webpage(
+            'https://www.floatplane.com/', None,
+            note='Fetching session cookie')
+        self.session_id = self._get_cookies(
+            'https://www.floatplane.com')['sails.sid'].value
+
+    def _real_extract(self, url):
+        post_id = self._match_id(url)
+
+        headers = {'Cookie': f"sails.sid={self.session_id}"}
+
+        ###
+        # Fetch post details
+        ###
+        post_api_url = f"https://www.floatplane.com/api/v3/content/post?id={post_id}"
+
+        post_metadata = self._download_json(
+            post_api_url, post_id, headers=headers, note='Fetching post details')
```

Comment:
```suggestion
        post_metadata = self._download_json(
            f'https://www.floatplane.com/api/v3/content/post?id={post_id}',
            post_id, headers=headers, note='Fetching post details')
```
Singe use variables should be inlined. Same below